designdocs/passkey_p256.md

```diff
@@ -0,0 +1,556 @@
+      let payload = {
+        passkeyPublicKey: Array.from(passkey_pk),
+        passkeySignature: Array.from(base64UrlToUint8Array(passkey_sig_tx)),
+        passkeyAuthenticator: Array.from(base64UrlToUint8Array(passkey_authenticator_data)),
+        passkeyClientDataJson: Array.from(base64UrlToUint8Array(
+          passkey_client_data)),
+        passkeyCall,
+      };
+
+      const passkey_payload = api.createType('PalletFrequencyTxPaymentPasskeyPayload', payload);
+
+      // Submit the transaction to the chain without signature
+      const tx = api.tx.{frequecyPassKeyProxy}.submitPasskeyTx(passkey_payload);
+      await tx.send()
+    ```
+
+4. **Backend Validation and Submission**:
+    - The backend can further validate the transaction payload before submitting it to the chain.
+    - The backend validates the `passkey_sig_tx` and the `passkey_authenticator_data` to ensure the authenticity of the transaction.
+    - For more secure operations, handling the transaction creation and submission on client side is recommended.
+
+### Security Considerations
+
+#### General
+
+- Never store or backup the seed phrase other than memory.
+- Prefer more client side data processing and handling, backend should only store the necessary data.
+- Choose vetted libraries, npm packages for handling cryptographic operations, such that signing with seed phrase, or handling of private keys is done securely.
+
+#### Front-end (client)
+
+- If key generation is done in front-end, it should ideally being done inside an isolated section such as iframe or Web Worker.
+- Generated Keypair should not get stored permanently (except for back up options) and removed as soon as it is not required.
+
+#### Backend
+
+- PassKey registration response should get verified which checks the random challenge. Random Challenge
+for registration is generated account Public Key (`account_pk`)
+- PassKey Login response should get verified which checks the random challenge
+- PassKey Transaction response should get verified which checks transaction related challenge.
+- Any provided Frequency account signature should get verified.
+
+#### On-chain
+
+- Preferably using an audited crate to support p256 operations. Currently, we are using `p256` crate
+  which is not audited.
+  - After investigating other libraries it looks like the only audited alternative is `Openssl` library which is only
+a wrapper around the Original C implementation and unfortunately it doesn't look like it is compatible with WASM.
+- If signature checks implemented **on_validate** are expensive, then this would open a vulnerability
+  surface for DOS attacks.
+
+## 6. Implementation
+
+### On-chain Implementation
+
+#### Extrinsic verification
+
+Currently, the Frequency chain primarily makes use of _SignedExtrinsic_. As we know, every signed extrinsic is validated by a list of SignedExtensions which each verify a different aspect of that extrinsic. The list of signed extensions includes:
```

Comment:
How much do we need to explain about Polkadot/Substrate? Should there be pointers to substrate docs?

```suggestion
Currently, the Frequency chain primarily makes use of _SignedExtrinsic_. Every signed extrinsic is validated by a list of SignedExtensions which each verify a different aspect of that extrinsic. The list of signed extensions includes:
```